api/src/main/java/gov/cms/ab2d/api/controller/BulkDataAccessAPI.java

```diff
@@ -189,15 +205,23 @@ private void logSuccessfulJobCreation(Job job) {
     }
 
     private ResponseEntity<Void> returnStatusForJobCreation(Job job) {
-        String statusURL = ServletUriComponentsBuilder.fromCurrentRequestUri().replacePath
-                (String.format(API_PREFIX + FHIR_PREFIX + "/Job/%s/$status", job.getJobUuid())).toUriString();
+        String statusURL = getUrl(API_PREFIX + FHIR_PREFIX + "/Job/" + job.getJobUuid() + "/$status");
         HttpHeaders responseHeaders = new HttpHeaders();
         responseHeaders.add("Content-Location", statusURL);
 
         return new ResponseEntity<>(null, responseHeaders,
                 HttpStatus.ACCEPTED);
     }
 
+    String getUrl(String ending) {
+        String url = ServletUriComponentsBuilder.fromCurrentRequestUri().replacePath(ending).toUriString();
+        url = url.replace(":80/", "/");
+        if (!shouldReplaceWithHttps()) {
+            return url;
+        }
+        return url.replace("http://", "https://");
```

Comment:
These lines duplicate those in `getCurrentUrl`; let's refactor into a common method.